File: student-service/src/main/java/kz/baltabayev/studentservice/model/service/impl/FacultyServiceImpl.java
package kz.baltabayev.studentservice.model.service.impl;

import kz.baltabayev.studentservice.model.entity.Faculty;
import kz.baltabayev.studentservice.model.service.FacultyService;
import kz.baltabayev.studentservice.repository.FacultyRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class FacultyServiceImpl implements FacultyService {

    private final FacultyRepository facultyRepository;

    @Override
    public List<Faculty> getAll() {
        return facultyRepository.findAll();
    }

    @Override
    public Faculty getById(Long id) {
        return facultyRepository.findById(id).orElseThrow(() -> new RuntimeException("Faculty not found"));
    }

    @Override
    public Faculty create(Faculty faculty) {
        return facultyRepository.save(faculty);
    }

    @Override
    public Faculty update(Faculty faculty) {
        return facultyRepository.save(faculty);
    }

    @Override
    public void deleteFacultyById(Long id) {
        facultyRepository.deleteById(id);
    }
}

File: faculty-department-service/src/main/kotlin/kz/baltabayev/facultydepartmentservice/service/impl/FacultyServiceImpl.kt
package kz.baltabayev.facultydepartmentservice.service.impl

import kz.baltabayev.facultydepartmentservice.exception.EntityNotFoundException
import kz.baltabayev.facultydepartmentservice.model.entity.Faculty
import kz.baltabayev.facultydepartmentservice.repository.FacultyRepository
import kz.baltabayev.facultydepartmentservice.service.FacultyService

class FacultyServiceImpl(
    private val facultyRepository: FacultyRepository
) : FacultyService {

    override fun save(faculty: Faculty) {
        facultyRepository.save(faculty)
    }

    override fun update(faculty: Faculty) {
        facultyRepository.save(faculty)
    }

    override fun delete(id: Long) {
        facultyRepository.deleteById(id)
    }

    override fun findById(id: Long): Faculty {
        return facultyRepository.findById(id)
            .orElseThrow { EntityNotFoundException(Faculty::class.java, id) }
    }
}